buildSrc/src/main/java/org/elasticsearch/gradle/DependenciesGraphTask.java

```diff
@@ -0,0 +1,182 @@
+                ModuleVersionIdentifier id = a.getModuleVersion().getId();
+                return id.getGroup() + ":" + id.getName() + "@" + id.getVersion();
+            })
+            .collect(Collectors.toSet());
+        for (final Dependency dependency : runtimeDependencies) {
+            final String id = dependency.getGroup() + ":" + dependency.getName();
+            final String versionedId = id + "@" + dependency.getVersion();
+            final StringBuilder packageString = new StringBuilder();
+            final StringBuilder nodeString = new StringBuilder();
+            if (compileOnlyArtifacts.contains(versionedId)) {
+                continue;
+            }
+            if (dependency instanceof ProjectDependency) {
+                continue;
+            }
+            packageString.append("{\"id\": \"")
+                .append(versionedId)
+                .append("\",\"info\": {\"name\": \"")
+                .append(id)
+                .append("\",\"version\": \"")
+                .append(dependency.getVersion())
+                .append("\"}}");
+            packages.add(packageString.toString());
+            nodeString.append("{\"nodeId\": \"")
+                .append(versionedId)
+                .append("\",\"pkgId\": \"")
+                .append(versionedId)
+                .append("\",\"deps\": []}");
+            nodes.add(nodeString.toString());
+            nodeIds.add("{\"nodeId\": \"" + versionedId + "\"}");
+        }
+        // We add one package and one node for each dependency, it suffices to check packages.
+        if (packages.size() > 0) {
+            final String projectName = "elastic/elasticsearch" + getProject().getPath();
+            final StringBuilder output = new StringBuilder();
+            output.append("{\"depGraph\": {\"schemaVersion\": \"1.2.0\",\"pkgManager\": {\"name\": \"gradle\"},\"pkgs\": [")
+                .append("{\"id\": \"")
+                .append(projectName)
+                .append("@0.0.0")
+                .append("\", \"info\": {\"name\": \"")
+                .append(projectName)
+                .append("\", \"version\": \"0.0.0\"}},")
+                .append(String.join(",", packages))
+                .append("],\"graph\": {\"rootNodeId\": \"")
+                .append(projectName)
+                .append("@0.0.0")
+                .append("\",\"nodes\": [")
+                .append("{\"nodeId\": \"")
+                .append(projectName)
+                .append("@0.0.0")
+                .append("\",\"pkgId\": \"")
+                .append(projectName)
+                .append("@0.0.0")
+                .append("\",\"deps\": [")
+                .append(String.join(",", nodeIds))
+                .append("]},")
+                .append(String.join(",", nodes))
+                .append("]}}}");
+            getLogger().debug("Dependency Graph: " + output.toString());
+            try (CloseableHttpClient client = HttpClients.createDefault()) {
```

Comment:
Since this task performs network operations, we should probably explicitly fail if a user attempts to run it in offline mode, [similarly to what we do for the `verifyVersions` task](https://github.com/elastic/elasticsearch/blob/27089e74602dc442656cf97ca4b29af79f724933/build.gradle#L148).

Reply:
Thanks, will do